apps/jubmoji-quest/src/components/powers/RedirectPower.tsx

```diff
@@ -4,17 +4,29 @@ import toast from "react-hot-toast";
 import { PowerContentProps } from "@/pages/powers/[id]";
 import { Button } from "../ui/Button";
 import { Prisma } from "@prisma/client";
+import { useState } from "react";
+import { ProvingState } from "jubmoji-api";
 
 const RedirectPower = ({ power, jubmojis }: PowerContentProps) => {
+  const [provingState, setProvingState] = useState<ProvingState>();
   const powerParams = power.powerParams as Prisma.JsonObject;
   const redirectUrl = powerParams.redirectUrl as string | undefined;
   const powerMutation = useRedirectPowerMutation();
 
+  const onUpdateProvingState = (newProvingState: ProvingState) => {
+    if (newProvingState.numProofsCompleted === newProvingState.numProofsTotal) {
+      setProvingState(undefined);
+    } else {
+      setProvingState(newProvingState);
+    }
+  };
+
   const onRedirect = async () => {
     try {
       const serializedProof = await powerMutation.mutateAsync({
         power,
         jubmojis,
+        onUpdateProvingState,
       });
 
       const url = new URL(redirectUrl!);
```

Comment:
Redirect URL needs to start with "https://" to work

Reply:
documented